spec/unit/hanami/routing/error_spec.rb

```diff
@@ -6,6 +6,8 @@
   it 'is parent to all custom exception' do
     expect(Hanami::Routing::Parsing::BodyParsingError.superclass).to eq(Hanami::Routing::Error)
     expect(Hanami::Routing::Parsing::UnknownParserError.superclass).to eq(Hanami::Routing::Error)
+    expect(Hanami::Routing::Middleware::BodyParsingError.superclass).to eq(Hanami::Routing::Error)
```

Comment:
We can't introduce new exceptions.

Imagine someone is doing:

```ruby
begin
  # ...
rescue Hanami::Routing::Parsing::BodyParsingError
end
```

Then suddenly the body parser raises `Hanami::Routing::Middleware::BodyParsingError` instead. The `rescue` block won't work anymore. This is a breaking change.